src/main/java/ratpack/zipkin/TracedParallelBatch.java

```diff
@@ -0,0 +1,34 @@
+package ratpack.zipkin;
+
+import brave.propagation.TraceContext;
+import java.util.Arrays;
+import ratpack.exec.Promise;
+import ratpack.exec.util.ParallelBatch;
+import ratpack.zipkin.internal.RatpackCurrentTraceContext.TraceContextTypeValue;
+
+public class TracedParallelBatch {
+
+  /**
+   * Add {@link TraceContext} propagation to {@link ParallelBatch#of(Promise[])}.
+   *
+   * Example usage:
+   *
+   * TracedParallelBatch
```

Comment:
Pulled in the ParallelBatch work that @edwardsoo-ss was working on.  Essentially the same thing except I made some helper methods to abstract away the `.execInit` stuff.